src/dfx/src/lib/extension/manager/install.rs

```diff
@@ -12,16 +15,39 @@ use std::io::Cursor;
 use std::os::unix::fs::PermissionsExt;
 
 impl ExtensionManager {
-    pub fn install_extension(&self, extension_name: &str) -> Result<(), ExtensionError> {
+    pub fn install_extension(
+        &self,
+        extension_name: &str,
+        external_manifest_url: Option<&str>,
+    ) -> Result<(), ExtensionError> {
         if self.get_extension_directory(extension_name).exists() {
             return Err(ExtensionError::ExtensionAlreadyInstalled(
                 extension_name.to_string(),
             ));
         }
 
-        let url = self.get_extension_download_url(extension_name)?;
-
-        let temp_dir = self.download_and_unpack_extension_to_tempdir(url)?;
+        let (download_url, checksum) = if let Some(external_manifest_url) = external_manifest_url {
+            let manifest = ExternalExtensionManifest::fetch(external_manifest_url)?;
+            let platform_arch = format!("{}-{}", std::env::consts::OS, std::env::consts::ARCH);
+            let binary_descriptor_manifest_entry = manifest
+                .find_extension(extension_name, &self.dfx_version_strip_semver())?
+                .get_binary_descriptor(&platform_arch)?;
+            (
+                binary_descriptor_manifest_entry.url.clone(),
+                binary_descriptor_manifest_entry.sha256.clone(),
+            )
+        } else {
+            let url = self.get_extension_download_url(extension_name)?;
+            // let checksum = self.get_extension_checksum(extension_name)?; // TODO where should we get the checksum from?
+            let checksum = String::new();
```

Comment:
This will break any attempts of installing extensions from `dfinity/dfx-extensions` repository (checksum simply won't match). 
I'll fix it in another PR.